src/firefly/test/edu/caltech/ipac/visualize/plot/HistogramTest.java

```diff
@@ -0,0 +1,362 @@
+package edu.caltech.ipac.visualize.plot;
+
+import nom.tam.fits.Fits;
+import nom.tam.fits.FitsException;
+import org.json.simple.JSONArray;
+import org.json.simple.JSONObject;
+import org.json.simple.parser.JSONParser;
+import org.json.simple.parser.ParseException;
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Test;
+import org.junit.Before;
+
+import java.io.*;
+
+/**
+ * Created by zhang on 10/4/16.
+ */
+public class HistogramTest {
+
+    private static final String TEST_DATA_ROOT = "firefly_test_data/";
+    private static final String TEST_DATA_PATH= TEST_DATA_ROOT+HistogramTest.class.getCanonicalName().replaceAll("\\.", "/")
+            .replace(HistogramTest.class.getSimpleName(), "") + File.separatorChar;
+    private static String fileName = "f3.fits";
+
+    /*
+     * 12 bit pixel image has 4096 colors
+     *
+     */
+    private float[] inData = null;
+    private static int HISTSIZ2 = 4096;    /* full size of hist array */
+    private Histogram hist;
+    private double dataMin=Double.MAX_VALUE;
+    private double dataMax=Double.MIN_VALUE;
+    private double delta = 1.010E-10;
+    double hbinSize;
+    private int[] histArray = new int[HISTSIZ2 + 1];
+    private int len = 5000;
+    private float[] data = new float[len];
+
+    @Before
+    /**
+     * An one dimensional array is created and it is used to run the unit test for Histogram's public methods
+     */
+    public void setUp() throws FitsException {
+
+
+       float c = 1000.0f;
+       for (int i=0; i<len; i++){
+            data[i]= (float) (c*Math.abs(Math.sin(i*Math.PI/180.0)) );
+        }
+        dataMin = 0.0;
+        dataMax = 1000.0;
+        hbinSize = (dataMax - dataMin)/HISTSIZ2;
+        histArray = buildHistArray(data, 0, hbinSize);
+
+        hist = new Histogram(data, dataMin, dataMax);
+
+    }
+    @After
+    /**
+     * Release the memories
+     */
+    public void tearDown() {
+        inData=null;
+        histArray=null;
+        data=null;
+    }
```

Comment:
TearDown is good specially if you have many test after this one (or before, junit test runner doesn't run in order...)